crates/topos-tce-broadcast/src/double_echo/mod.rs

```diff
@@ -435,20 +446,23 @@ impl DoubleEcho {
             }
 
             let echo_missing = self
-                .params
-                .echo_sample_size
+                .subscribers
+                .get_subscribers()
```

Comment:
You're comparing two things that doesn't fit together.
`Subscribers` are people interested into receiving your messages (Echo/Ready), `Subscriptions` are peers that you want to receive message from. At some point the `DoubleEcho` will not have `subscribers` anymore.
You need to base the calculation on `network_size - current_subscription`, the current subscriptions are reduced when receiving a message upon reaching the threshold.

Example:

```
---- state ----
current node: peer1
all threshold: 2
network_size: 10 - 1 (peer1 is you)
Echo Subscribers: [ peer2, peer3, peer4, peer5, ..., peer10 ]
Echo Subscriptions: [ peer2, peer3, peer4, peer5, ..., peer10 ]
----

-> RECV: Echo from peer2
    -> Remove peer2 from Echo Subscriptions

Echo Subscriptions: [ peer3, peer4, peer5, ..., peer10 ]

Threshold ? network_size - len(Echo Subscriptions) -> 9 - 8 = 1 : NO

----

-> RECV: Echo from peer4
    -> Remove peer4 from Echo Subscriptions

Echo Subscriptions: [ peer3, peer5, ..., peer10 ]

Threshold ? network_size - len(Echo Subscriptions) -> 9 - 7 = 2 : YES

```

Reply:
Thank you so much for the explanation, now it finally clicked. Makes sense 👍